src/Acuminator/Acuminator.Utilities/Roslyn/Semantic/PXGraph/Utils/GraphActionSymbolUtils.cs

```diff
@@ -0,0 +1,269 @@
+						.ForEach(handler => handlersCollection.Add(handler.Symbol.Name, handler));
+			}
+		}
+
+		private static IEnumerable<(MethodDeclarationSyntax Node, IMethodSymbol Symbol)> GetActionHandlersFromGraphImpl(
+																this ITypeSymbol graph, IDictionary<string, ActionInfo> actionsByName,
+																PXContext pxContext, CancellationToken cancellation, bool inheritance)
+		{
+			if (inheritance)
+			{
+				return graph.GetBaseTypesAndThis()
+							.TakeWhile(baseGraph => !baseGraph.IsGraphBaseType())
+							.Reverse()
+							.SelectMany(baseGraph => GetActionHandlersFromGraphOrGraphExtension(baseGraph, actionsByName, pxContext, cancellation));
+			}
+			else
+			{
+				return GetActionHandlersFromGraphOrGraphExtension(graph, actionsByName, pxContext, cancellation);
+			}
+		}
+
+		private static IEnumerable<(MethodDeclarationSyntax Node, IMethodSymbol Symbol)> GetActionHandlersFromGraphOrGraphExtension(
+															this ITypeSymbol graphOrExtension, IDictionary<string, ActionInfo> actionsByName, 
+															PXContext pxContext, CancellationToken cancellation)
+		{
+			IEnumerable<IMethodSymbol> handlers = from method in graphOrExtension.GetMembers().OfType<IMethodSymbol>()
+												  where method.IsValidActionHandler(pxContext) && actionsByName.ContainsKey(method.Name)
+												  select method;
+												  											 
+			foreach (IMethodSymbol handler in handlers)
+			{
+				cancellation.ThrowIfCancellationRequested();
+
+				SyntaxReference reference = handler.DeclaringSyntaxReferences.FirstOrDefault();
+
+				if (reference?.GetSyntax(cancellation) is MethodDeclarationSyntax declaration)
+				{
+					yield return (declaration, handler);
+				}		
+			}
+		}
+		#endregion
+
+		private static IEnumerable<GraphOverridableItem<T>> GetActionInfoFromGraphExtension<T>(ITypeSymbol graphExtension, PXContext pxContext,
+															Action<GraphOverridableItemsCollection<T>, ITypeSymbol> addGraphActionInfo,
+															Action<GraphOverridableItemsCollection<T>, ITypeSymbol> addGraphExtensionActionInfo)
+		{
+			var empty = Enumerable.Empty<GraphOverridableItem<T>>();
+
+			if (!graphExtension.InheritsFrom(pxContext.PXGraphExtensionType) || !graphExtension.BaseType.IsGenericType)
+				return Enumerable.Empty<GraphOverridableItem<T>>();
+
+			var actionsByName = new GraphOverridableItemsCollection<T>();
+			INamedTypeSymbol baseType = graphExtension.BaseType;
+			ITypeSymbol graphType = baseType.TypeArguments[baseType.TypeArguments.Length - 1];
+
+			if (!graphType.IsPXGraph(pxContext))
+				return empty;
+
+			addGraphActionInfo(actionsByName, graphType);
+
+			if (baseType.TypeArguments.Length >= 2 && !AddInfoFromBaseExtensions())
+				return empty;
+
+			addGraphExtensionActionInfo(actionsByName, graphExtension);
+			return actionsByName.Items;
+
+
+			//-----------------------------------------------------------------Local Function----------------------------------------------
```

Comment:
I think this comment looks ugly. What do you think, can we just keep local functions in the end of methods without any comment?

Reply:
It slightly helps me to distinguish local methods but I don't mind deleting them